client/blocks/upe/hooks.js

```diff
@@ -1,28 +1,251 @@
 
+/**
+ * Handles the Block Checkout onCheckoutSuccess event.
+ *
+ * Confirms the payment intent which was created on server and is now ready to be confirmed. The intent ID is passed in the paymentDetails object via the
+ * redirect arg which will be in the following format: #wc-stripe-confirm-pi/si:{order_id}:{client_secret}:{nonce}
+ *
+ * @param {*} api               The api object.
+ * @param {*} stripe            The Stripe object.
+ * @param {*} elements          The Stripe elements object.
+ * @param {*} onCheckoutSuccess The onCheckoutSuccess event.
+ * @param {*} emitResponse      Various helpers for usage with observer.
+ * @param {*} shouldSavePayment Whether or not to save the payment method.
+ */
 export const usePaymentCompleteHandler = (
 	api,
 	stripe,
 	elements,
-	onCheckoutAfterProcessingWithSuccess,
+	onCheckoutSuccess,
 	emitResponse,
 	shouldSavePayment
 ) => {
 	// Once the server has completed payment processing, confirm the intent of necessary.
 	useEffect(
 		() =>
-			onCheckoutAfterProcessingWithSuccess(
-				( { processingResponse: { paymentDetails } } ) =>
-					confirmCardPayment(
-						api,
-						paymentDetails,
-						emitResponse,
-						shouldSavePayment
-					)
+			onCheckoutSuccess( ( { processingResponse: { paymentDetails } } ) =>
+				confirmCardPayment(
+					api,
+					paymentDetails,
+					emitResponse,
+					shouldSavePayment
+				)
 			),
 		// not sure if we need to disable this, but kept it as-is to ensure nothing breaks. Please consider passing all the deps.
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 		[ elements, stripe, api, shouldSavePayment ]
 	);
 };
+
+/**
+ * Handles the Block Checkout onCheckoutFail event.
+ *
+ * Displays the error message returned from server in the paymentDetails object in the PAYMENTS notice context container.
+ *
+ * @param {*} api            The api object.
+ * @param {*} stripe         The Stripe object.
+ * @param {*} elements       The Stripe elements object.
+ * @param {*} onCheckoutFail The onCheckoutFail event.
+ * @param {*} emitResponse   Various helpers for usage with observer.
+ */
+export const usePaymentFailHandler = (
+	api,
+	stripe,
+	elements,
+	onCheckoutFail,
+	emitResponse
+) => {
+	useEffect(
+		() =>
+			onCheckoutFail( ( { processingResponse: { paymentDetails } } ) => {
+				return {
+					type: 'failure',
+					message: paymentDetails.errorMessage,
+					messageContext: emitResponse.noticeContexts.PAYMENTS,
+				};
+			} ),
+		[
+			elements,
+			stripe,
+			api,
+			onCheckoutFail,
+			emitResponse.noticeContexts.PAYMENTS,
+		]
+	);
+};
+
+/**
+ * Handles rendering the Block Checkout Stripe Link payment method.
+ *
+ * @param {*} api                  The api object.
+ * @param {*} elements             The Stripe elements object.
+ * @param {*} paymentMethodsConfig The payment methods config object. Used to determine if Stripe Link is enabled.
+ */
+export const useStripeLink = ( api, elements, paymentMethodsConfig ) => {
```

Comment:
> Or will that come in a later PR when we remove the old UPE implementation?

Yeah that's correct. That `fields.js` file is the old file responsible for rendering the Unified PE elements and is no longer needed. I've removed it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2784/commits/ad0a335bc249beb31ee54a4c229c5de9d4904dc0

The `confirm-upe-payment.js` file is also no longer used but I can't tell if that's only because I haven't implemented the other payment methods yet.